discovery.js

```
@@ -121,18 +132,62 @@ DiscoveryClient.prototype.reconnect = function (onComplete) {
   });
 };
 
+function _doAbort(request, callback) {
+  if (request) {
+    request.on('abort', function() {
+      if (callback) {
+        callback();
+      }
+    });
+    request.on('error', function(err){
+      if(callback){
+        callback(err);
+      }
+    });
+    request.abort();
+  }
+}
+
+/* Allow the discovery client to cleanly disconnect and shut down all handles */
+DiscoveryClient.prototype.disconnect = function(disconnectCB) {
+  var disco = this;
+  clearTimeout(disco._backoffTimeout);
+  clearInterval(disco._announceInterval);
+  var _onlyOneRequest;
+  var _gotErr = false;
+
+  //if both requests are valid then we have to wait for both abort events
+  if (disco._request && disco._pollRequest) {
+    _onlyOneRequest = false;
+  } else {
+    _onlyOneRequest = true;
+  }
+
+  var checkForCB = function(err) {
```

Does this mean if both aborts happen and both error, one error is swallowed?